pkg/controller/master/supportbundle/controller.go

```diff
@@ -70,12 +73,48 @@ func (h *Handler) OnSupportBundleChanged(key string, sb *harvesterv1.SupportBund
 	case types.StateGenerating:
 		logrus.Debugf("[%s] support bundle is being generated", sb.Name)
 		return h.checkManagerStatus(sb)
+	case types.StateError, types.StateReady:
+		return h.checkExistTime(sb)
 	default:
 		logrus.Debugf("[%s] noop for state %s", sb.Name, sb.Status.State)
 		return sb, nil
 	}
 }
 
+// checkExistTime checks if the support bundle has been updated for more than supportBundleExistTimeLimit minutes
+// If yes, that means the support bundle file is not be retrieved, and it should be deleted.
+func (h *Handler) checkExistTime(sb *harvesterv1.SupportBundle) (*harvesterv1.SupportBundle, error) {
+	t, err := time.Parse(time.RFC3339, harvesterv1.SupportBundleInitialized.GetLastUpdated(sb))
+	if err != nil {
+		logrus.Debugf("[%s] fail to parse %s", sb.Name, sb.Status.Conditions[0].LastUpdateTime)
```

Comment:
we should probably use `harvesterv1.SupportBundleInitialized.GetLastUpdated(sb)` in the error message, rather than assuming this will always be the first condition. This should ensure that we future proof against any more conditions being added.

Reply:
I forgot to modify usage in debug message.